great_expectations/validator/metrics_calculator.py

```diff
@@ -0,0 +1,400 @@
+
+        Returns:
+            Return Dictionary with requested metrics resolved, with metric_name as key and computed metric as value.
+        """
+        resolved_metrics: Dict[Tuple[str, str, str], Any] = self.compute_metrics(
+            metric_configurations=list(metrics.values()),
+        )
+
+        return {
+            metric_configuration.metric_name: resolved_metrics[metric_configuration.id]
+            for metric_configuration in metrics.values()
+        }
+
+    def compute_metrics(
+        self,
+        metric_configurations: List[MetricConfiguration],
+    ) -> Dict[Tuple[str, str, str], Any]:
+        """
+        Args:
+            metric_configurations: List of desired MetricConfiguration objects to be resolved.
+
+        Returns:
+            Dictionary with requested metrics resolved, with unique metric ID as key and computed metric as value.
+        """
+        graph: ValidationGraph = ValidationGraph()
+
+        metric_configuration: MetricConfiguration
+        for metric_configuration in metric_configurations:
+            provider_cls, _ = get_metric_provider(
+                metric_configuration.metric_name, self._execution_engine
+            )
+
+            self._get_default_domain_kwargs(
+                metric_provider_cls=provider_cls,
+                metric_configuration=metric_configuration,
+            )
+            self._get_default_value_kwargs(
+                metric_provider_cls=provider_cls,
+                metric_configuration=metric_configuration,
+            )
+
+            self.build_metric_dependency_graph(
+                graph=graph,
+                metric_configuration=metric_configuration,
+            )
+
+        resolved_metrics: Dict[Tuple[str, str, str], Any] = {}
+
+        # updates graph with aborted metrics
+        aborted_metrics_info: Dict[
+            Tuple[str, str, str],
+            Dict[str, Union[MetricConfiguration, Set[ExceptionInfo], int]],
+        ] = self.resolve_validation_graph(
+            graph=graph,
+            metrics=resolved_metrics,
+        )
+
+        if aborted_metrics_info:
+            logger.warning(
+                f"Exceptions\n{str(aborted_metrics_info)}\noccurred while resolving metrics."
+            )
+
+        return resolved_metrics
+
+    def build_metric_dependency_graph(
+        self,
+        graph: ValidationGraph,
+        metric_configuration: MetricConfiguration,
+        runtime_configuration: Optional[dict] = None,
+    ) -> None:
+        """Obtain domain and value keys for metrics and proceeds to add these metrics to the validation graph
+        until all metrics have been added."""
+
+        metric_impl = get_metric_provider(
+            metric_configuration.metric_name, execution_engine=self._execution_engine
+        )[0]
+        metric_dependencies = metric_impl.get_evaluation_dependencies(
```

Comment:
Not actionable for this PR, but in this method, `get_evaluation_dependencies`, and `resolve_validation_graph` is where I think we should look into implementing the logic for conditional evaluation dependencies. We could return multiple evaluation dependency graphs here, and then determine which is preferable during the resolution stage depending on other requested metrics. I will write this up in a document for review.